src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever.java

```diff
@@ -77,172 +74,118 @@ public ElasticsearchLogStorageRetriever(String elasticsearchUrl, Credentials ela
         if (StringUtils.isBlank(elasticsearchUrl)) {
             throw new IllegalArgumentException("Elasticsearch url cannot be blank");
         }
-        this.elasticsearchUrl = elasticsearchUrl;
         if (StringUtils.isBlank(indexPattern)) {
             throw new IllegalArgumentException("Elasticsearch Index Pattern cannot be blank");
         }
         this.indexPattern = indexPattern;
 
-        this.credentialsProvider = new BasicCredentialsProvider();
-        this.credentialsProvider.setCredentials(AuthScope.ANY, elasticsearchCredentials);
-    }
+        BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY, elasticsearchCredentials);
 
-    @Nonnull
-    @Override
-    public ByteBuffer overallLog(@Nonnull String traceId, @Nonnull String spanId) throws IOException {
-        return retrieveTraceLogs(traceId, spanId);
+        RestClient restClient = RestClient.builder(HttpHost.create(elasticsearchUrl))
+            .setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider))
+            .build();
+        RestClientTransport elasticsearchTransport = new RestClientTransport(restClient, new JacksonJsonpMapper());
+        this.elasticsearchClient = new ElasticsearchClient(elasticsearchTransport);
     }
 
     @Nonnull
     @Override
-    public ByteBuffer stepLog(@Nonnull String traceId, @Nonnull String spanId) throws IOException {
-        return retrieveTraceLogs(traceId, spanId);
+    public LogsQueryResult overallLog(@Nonnull String traceId, @Nonnull String spanId, @Nullable LogsQueryContext logsQueryContext) throws IOException {
+
+        ByteBuffer byteBuffer = new ByteBuffer();
+        String newScrollId;
+        Charset charset = StandardCharsets.UTF_8;
+        boolean complete;
+        if (logsQueryContext == null) {
+            SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(this.indexPattern)
+                .scroll(SCROLL_TTL)
+                .size(PAGE_SIZE)
+                .sort(sortBuilder -> sortBuilder.field(fieldBuilder -> fieldBuilder.field(TIMESTAMP).order(SortOrder.Asc)))
+                .query(queryBuilder ->
+                    queryBuilder.match(
+                        matchQueryBuilder -> matchQueryBuilder.field(TRACE_ID).query(
+                            fieldValueBuilder -> fieldValueBuilder.stringValue(traceId))))
+                // .fields() TODO narrow down the list fields to retrieve
+                .build();
+            SearchResponse<JsonObject> searchResponse = this.elasticsearchClient.search(searchRequest, JsonObject.class);
```

Comment:
@swallez what's the best data structure to get a JSON document or something similar? `JsonObject` is rejected.